base/src/main/scala/co/uproot/abandon/Report.scala

```diff
@@ -102,44 +102,75 @@ object Reports {
     BalanceReport(leftRender, rightRender, "%" + leftAmountWidth + ".2f" format leftTotal, "%" + rightAmountWidth + ".2f = %s" format (rightTotal, totalStr))
   }
 
+
+  /**
+    * Formats int date used for grouping to text, based on it's resolution
+    *
+    * @param groupingDate
+    * @return Date as string with correct resolution
+    */
   private def formatGroupingDate(groupingDate: Int) = {
     val date = Date.fromInt(groupingDate)
     if (date.hasDayResolution) {
       date.formatISO8601Ext
     }
-    else {
+    else if (date.hasMonthResolution) {
       s"${date.year} / ${Helper.monthLabels(date.month - 1)}"
     }
+    else {
+      s"${date.year}"
+    }
   }
 
   def registerReport(state: AppState, reportSettings: RegisterReportSettings): Seq[RegisterReportGroup] = {
-    val postGroups = state.accState.postGroups.filter(_.children.exists(c => reportSettings.isAccountMatching(c.name.fullPathStr)))
-    val monthlyGroups = postGroups.groupBy(d => d.date.toIntYYYYMM).toSeq.sortBy(_._1)
+    val txns = state.accState.postGroups.filter(_.children.exists(c => reportSettings.isAccountMatching(c.name.fullPathStr)))
+
+    /**
+      * Grouping is done by truncated int numbers. The end result (groups) is
+      * Seq[(String, Seq[PostGroup])] so we need mapping from truncated intDate -> String
```

Comment:
That's one option. However, probably most of the cases sorting is done by year, month or day. By doing this by Int should be faster than purely string based sort. However, I don't have any specific numbers....